proof/invariant-abstract/AARCH64/ArchVCPU_AI.thy

```diff
@@ -116,12 +116,14 @@ crunch vcpu_restore, do_machine_op, vcpu_save_reg, vgic_update, save_virt_timer,
   (wp: valid_cur_vcpu_lift crunch_wps simp: active_cur_vcpu_of_def)
 
 lemma switch_vcpu_valid_cur_vcpu_cur_thread_update[wp]:
-  "\<lbrace>arch_tcb_at (\<lambda>itcb. itcb_vcpu itcb = v) t\<rbrace>
+  "\<lbrace>\<lambda>s. arch_tcb_at (\<lambda>itcb. itcb_vcpu itcb = v) t s\<rbrace>
```

Comment:
Was this eta expansion needed? The two terms should be equivalent. It's possible that the rule `hoare_lift_Pf` only matches agains the eta-expanded form (even though unification should be modulo eta). In that case, we could try to use the `wps` method instead of the explicit rule application.